PromisesObjC.podspec

```diff
@@ -23,6 +23,7 @@ Pod::Spec.new do |s|
   s.public_header_files = "Sources/#{s.module_name}/include/**/*.h"
   s.private_header_files = "Sources/#{s.module_name}/include/FBLPromisePrivate.h"
   s.source_files = "Sources/#{s.module_name}/**/*.{h,m}"
+  s.resource = "Sources/Resources/PrivacyInfo.xcprivacy"
```

Comment:
I _think_ this should instead be packaged via the `resource_bundle` API instead. I'm experimenting to see how Firebase's release tooling handles `resource`.

Reply:
Thanks for looking into it. I did see other podspecs where this was included as a `resource`:
https://github.com/treasure-data/td-ios-sdk/blob/master/TreasureData-iOS-SDK.podspec
https://github.com/getsentry/sentry-cocoa/blob/main/Sentry.podspec

but I'll wait for your confirmation.

Reply:
I've seen all 3 combinations work:
* `s.resource`
* `s.resources`
* s.resource_bundle`

Since this is a singular file for now, I'd keep it as `resource` (although `resource_bundle` should work too).

Reply:
Using `resource_bundle` is the correct path here. The issue with `resource[s]` surfaces when clients build the pod with _static linking_ enabled (`use_frameworks! :linkage => :static` in their Podfile). The pod's `PrivacyInfo.xcprivacy` gets thrown into the root of the app bundle, where it may collide with other manifests from the app or the app's other direct/transitive dependencies. Secondly– and I'm unsure if/how this will be enforced– the manifest won't be name spaced to the pod that owns it, so the app's resulting Privacy Report won't say _which_  SDK the information maps to. Examples below...

### App bundle when using `resource[s]`
```
PrivacyAppCocoaPods.app
├── Info.plist
├── PkgInfo
├── PrivacyAppCocoaPods
├── PrivacyInfo.xcprivacy   ← 💥 Will collide with other manifests. Not name spaced to pod.
└── _CodeSignature
    └── CodeResources
```

### App bundle when using `resource_bundle`
```
PrivacyAppCocoaPods.app
├── PromisesObjC_Privacy.bundle
│   ├── Info.plist
│   └── PrivacyInfo.xcprivacy   ← ✅ Not in the root of the app bundle. Name spaced to pod.
├── Info.plist
├── PkgInfo
├── PrivacyAppCocoaPods
└── _CodeSignature
    └── CodeResources
```

I suggest doing something like this:

```suggestion
  s.resource_bundle = {
    "#{s.name}_Privacy" => 'Sources/Resources/PrivacyInfo.xcprivacy'
  }
```

The above suggestion will support both the dynamic linking case and the static linking case.

Reply:
@ncooke3 Super useful to know, thanks for the thorough explanation.